spec/firebase_test_lab_result_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'spec_helper'
+
+EMPTY_FIREBASE_TEST_LOG_PATH = File.join(__dir__, 'test-data', 'empty.json')
+FAILED_FIREBASE_TEST_LOG_PATH = File.join(__dir__, 'test-data', 'firebase', 'failed-firebase-test-lab-run.log')
+INVALID_TEST_LOG_PATH = 'foo'.freeze
+
+describe Fastlane::FirebaseTestLabResult do
+  let(:empty_test_log) { described_class.new(log_file_path: EMPTY_FIREBASE_TEST_LOG_PATH) }
+  let(:failed_test_log) { described_class.new(log_file_path: FAILED_FIREBASE_TEST_LOG_PATH) }
+
+  describe 'initialize' do
+    it 'raises for an invalid file path' do
+      expect { described_class.new(log_file_path: INVALID_TEST_LOG_PATH) }.to raise_exception 'No log file found at path foo'
+    end
+  end
+
+  describe 'success?' do
+    it 'returns false for failure' do
+      expect(failed_test_log.success?).to be false
```

Comment:
💥 🤔  Given my previous comment on this method using `.none? { !include?("Failed") }` being incorrect… I'm surprised this test passed.

This maybe means that instead of just blindly fixing the `!include?` in the implementation, we should probably try to go the red-green-refactoring / TDD-oriented route for fixing this properly and ensuring coverage is correct:
 - Create a new test and make sure it fails with the current wrong implementation of success?`, so that we can be sure that such red herring in result log files would be properly caught
 - Then fix the implementation
 - Then validate that test turns green

Otherwise if we just fix the implementation without fixing the test that means that the test won't catch any future failure during a potential future refactoring, or even incorrect implementations failing to return the expected result when tested against real-world scenarios and log files.

Reply:
ff89bd09a552bfebe2831404aaa9dcb4329f3243 adds a test for the `true` case